### ISSUE TITLE
Can't launch/operate firefox via geckodriver using Windows Task Scheduler 

### ISSUE THREAD

**Issue by bhyman** [mozilla/geckodriver#1134]
## System
* Version: 0.19.1
* Platform: Windows Server 2012 x64
* Firefox: 59.0.0.6589 (64bit) (the 1/15/18 build from https://ftp.mozilla.org/pub/firefox/nightly/2018/01/2018-01-15-10-00-56-mozilla-central/firefox-59.0a1.en-US.win64.zip ... issues experienced on 57.0.2 too)
* Selenium: 3.8.0, Python bindings

## Testcase
Save the following in a .py file and set it up via Task Scheduler as a scheduled task. Trigger it manually or with a time-based trigger.

```
import os
import datetime as dt
from selenium import webdriver
from selenium.webdriver.firefox.options import Options as FirefoxOptions
from selenium.webdriver.firefox.firefox_binary import FirefoxBinary


class Firefox(object):
    def __init__(self, download_dir):

        profile = webdriver.FirefoxProfile()

        profile.set_preference("browser.download.dir", download_dir)
        profile.set_preference("browser.download.folderList", 2)
        profile.set_preference("security.sandbox.content.level",4) #suggested from https://github.com/mozilla/geckodriver/issues/1095
        
        binary = FirefoxBinary(firefox_path=r"C:\Program Files\mozilla_firefox-59.0a1.en-US.win64\firefox.exe")

        options = FirefoxOptions()
        options.add_argument('-headless')
        options.log.level = "trace"

        self.driver = webdriver.Firefox(firefox_profile=profile, firefox_binary=binary, firefox_options=options, executable_path=r"C:\bin\misc\drivers\geckodriver.exe", log_path = 'geckodriver' + dt.datetime.today().strftime('%m%d%H%M%S') + '.log')

download_dir=r"downloads"
driver = Firefox(download_dir).driver
driver.get('https://www.google.com')
```

This happens regardless of whether Firefox is running in headless mode or not. Also, my particular use case involves a custom profile (a modified handlers.json file is the only difference), but the crash occurs even with a default profile, so that's what's exhibited here.

This seems to be a recurrence/combination of issues #1095 (note that I tried the fix therein unsuccessfully), #1089, and #1087; hoping I can provide some more detail or resolution than was accomplished in those. Also looks to be more of a firefox issue than a geckodriver one ... let me know if I should direct this elsewhere.

## Stacktrace
None - Task Scheduler doesn't provide serviceable logs, and the underlying script fails silently once geckodriver.exe/firefox.exe crashes.

## Trace-level log
[geckodriver0116105312.log](https://github.com/mozilla/geckodriver/files/1636100/geckodriver0116105312.log)

## Firefox dumps
[minidumps.zip](https://github.com/mozilla/geckodriver/files/1636105/minidumps.zip)

**Comment by whimboo** [comment]
Yes, this is a bug in Firefox and not geckodriver. Given your attached minidump files (thanks a lot for those!) I was able to find https://bugzilla.mozilla.org/show_bug.cgi?id=1400637.

Here is your crash report: https://crash-stats.mozilla.com/report/index/740876c8-0479-4b55-b2d1-e32bb0180117

Without having you to read all the comments, here some questions:

* Are you using an antivirus software? If yes, does it help to turn it off (at least the web content protection)?
* Could you change your test and use different values for the preference `security.sandbox.content.level` through the capabilities? Not sure what the maximum currently is for Windows. Maybe 4 or 5 (level 3 should work).
* Do you also see this crash when you do not let it run through a Windows task?

I will have to close this issue here given that it is not geckodriver related but please get back to us with the above info. Maybe you can even attach the output from `about:support`. You could also setup an account on Bugzilla if you want, and directly reply there.

Thanks again for this very helpful report! Lets see if we can get this high volume crasher fixed by your input.

**Comment by bhyman** [comment]
Thanks, whimboo. I just made a direct reply in Bugzilla -- per what I said there, this is now working fine in 59.0a1 and 60.0a1. Mystified as to why it was broken last week and functional this week. Oh well. I'll keep an eye out and will make further additions to Bugzilla as necessary. Appreciate your help!

**Comment by whimboo** [comment]
@bhyman great to hear! And thanks a lot. Please also always notify us in case you find something else.

**Comment by whimboo** [comment]
Ok, some updates here. As it looks like this crash should be fixed with the upcoming Firefox 60.0 beta release, and is already fixed on Nightly. The reason is that Firefox were processing native events in the content processes. This has been turned off on Windows with the preference dom.ipc.useNativeEventProcessing.content set to false.

@johno1985 and @lepik1108 as you can see this in issue #1177 can you both please try to run Nightly with the preferences turned on? Does this crash in that case, and doesn't crash with the pref turned off?
